src/phoenix/server/api/types/pagination.py

```diff
@@ -56,6 +60,73 @@ class Edge(Generic[GenericType]):
 CURSOR_PREFIX = "connection:"
 
 
+class SortableFieldType(Enum):
+    STRING = auto()
+    INT = auto()
+    FLOAT = auto()
+    DATETIME = auto()
+
+
+@dataclass
+class SortableField:
+    type: SortableFieldType
+    value: SortableFieldValue
+
+    def stringify_value(self) -> str:
+        if isinstance(self.value, str):
+            return self.value
+        if isinstance(self.value, (int, float)):
+            return str(self.value)
+        if isinstance(self.value, datetime):
+            return self.value.isoformat()
+        assert_never(self.type)
+
+    @classmethod
+    def from_stringified_value(
+        cls, type: SortableFieldType, stringified_value: str
+    ) -> "SortableField":
+        value: SortableFieldValue
+        if type is SortableFieldType.STRING:
+            value = stringified_value
+        elif type is SortableFieldType.INT:
+            value = int(stringified_value)
+        elif type is SortableFieldType.FLOAT:
+            value = float(stringified_value)
+        elif type is SortableFieldType.DATETIME:
+            value = datetime.fromisoformat(stringified_value)
+        else:
+            assert_never(type)
+        return cls(type=type, value=value)
+
+
+@dataclass
+class NodeIdentifier:
+    rowid: int
+    sortable_field: Optional[SortableField] = None
+
+    _DELIMITER: ClassVar[str] = ":"
+
+    def to_cursor(self) -> Cursor:
+        cursor_components = [str(self.rowid)]
+        if (sortable_field := self.sortable_field) is not None:
+            cursor_components.extend([sortable_field.type.name, sortable_field.stringify_value()])
+        return base64.b64encode(self._DELIMITER.join(cursor_components).encode()).decode()
+
+    @classmethod
+    def from_cursor(cls, cursor: Cursor) -> "NodeIdentifier":
+        decoded = base64.b64decode(cursor).decode()
+        rowid_string = decoded
+        sortable_field = None
+        if (first_delimiter_index := decoded.find(cls._DELIMITER)) > -1:
+            rowid_string = decoded[:first_delimiter_index]
+            second_delimiter_index = decoded.index(cls._DELIMITER, first_delimiter_index + 1)
+            sortable_field = SortableField.from_stringified_value(
+                type=SortableFieldType[decoded[first_delimiter_index + 1 : second_delimiter_index]],
+                stringified_value=decoded[second_delimiter_index + 1 :],
+            )
+        return cls(rowid=int(rowid_string), sortable_field=sortable_field)
```

Comment:
probably needs docs with examples just so the packing and unpacking of the cursor is a bit more clear